File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxLen = 0;
        unordered_set<char> set;

        int l = 0;
        for(int r = 0;r < s.size(); r++)
        {
            while(set.count(s[r]) > 0)
            {
                set.erase(s[l]);
                l++;
            }
            
            set.insert(s[r]);
            int len = r - l + 1;
            maxLen = max(len, maxLen);
            
        }

        return maxLen;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        int n = s.Length;
        if(n < 2)
            return n;
        HashSet<char> set = new HashSet<char>();
        int res = 0;
        int l = 0;
        set.Add(s[l]);
        for(int r = 1;r < n; r++)
        {
            while(set.Contains(s[r]))
            {
                set.Remove(s[l]);
                l++;
            }

            set.Add(s[r]);

            int len = r - l + 1;
            res = Math.Max(res, len);
        }

        return res;
    }
}